pkg/statedb2/metrics.go

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package statedb2
+
+import (
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+)
+
+type Metrics struct {
+	// How long a read transaction was held.
+	WriteTxnDuration metric.Vec[metric.Observer]
+	// How long it took to acquire a write transaction for all tables.
+	WriteTxnAcquisition metric.Vec[metric.Observer]
+	// How long writers were blocked while waiting to acquire a write transaction for a specific table.
+	TableContention metric.Vec[metric.Gauge]
```

Comment:
We already have a histogram above it which measures how long it takes for all locks in a transaction to be acquired, which should cover most information. The table contention metric is just slightly more specific since it is per table, using a gauge was an attempt to keep the amount of metric data down. This is just 1 value per table, histograms have lots of values per label.